csrc/executor.cpp

```diff
@@ -1315,35 +1314,13 @@ KernelArgumentHolder FusionExecutor::inferOutputSizes(
           fusion->outputs()[out_i]->isA<TensorView>(),
           "Cannot allocate outputs that are not tensors.");
       auto output_tv = fusion->outputs()[out_i]->as<TensorView>();
-
-      auto alias_it = std::find_if(
-          output_to_input_aliases.begin(),
-          output_to_input_aliases.end(),
-          [&](const auto& output_to_input) {
-            return output_to_input.first == (int)out_i;
-          });
-
-      if (alias_it != output_to_input_aliases.end()) {
-        // When aliasing output to an input, we do not need to allocate a new
-        // output but still need to push an entry.
-        ret.push(PolymorphicValue(0L));
-      } else {
-        const auto& [sizes, strides] = inferShapeOfOutput(output_tv, expr_eval);
-        const auto dtype = (output_tv->dtype() == DataType::Index)
-            ? data_type_to_aten(arg_index_type)
-            : data_type_to_aten(output_tv->dtype());
-        ret.pushTensorProxy(sizes, strides, dtype);
-      }
+      const auto& [sizes, strides] = inferShapeOfOutput(output_tv, expr_eval);
+      const auto dtype = (output_tv->dtype() == DataType::Index)
+          ? data_type_to_aten(arg_index_type)
+          : data_type_to_aten(output_tv->dtype());
+      ret.pushTensorProxy(sizes, strides, dtype);
```

Comment:
does this mean we are allocating memory for forwarded outputs then? We need to double check if the integration code is still blindly allocate buffer for every output, regardless of its alias status.

The old trick here is that we put a dummy output there so allocation wouldn't create any buffer there.

Reply:
I agree this is tricky to read and I'm onboard with us removing the code here.

Since we are at this topic, we can probably refactor the output allocation code and properly use aliased buffer there as well.

Reply:
This is where we skip this place holder in the existing code. https://github.com/NVIDIA/Fuser/blob/main/csrc/kernel_cache.cpp#L198C1-L198C1